### DISCUSSION TITLE
SpotifyPlus Card UI - Beta Test

### DISCUSSION THREAD

**Discussion by thlucas1** [thlucas1/spotifyplus_card#8] [Announcements]
The beta test period of the [SpotifyPlus Card plugin](https://github.com/thlucas1/spotifyplus_card) has opened, and I am looking for volunteers to put it through it's paces to shake out any bugs that might be hiding in its depths.

## Requirements
- A [Spotify](https://open.spotify.com/) Premium account is required.
- [SpotifyPlus integration >= 1.0.61](https://github.com/thlucas1/homeassistantcomponent_spotifyplus/wiki) must be installed an configured.
- Home Assistant 2024.10+ release.

## Getting Started
- make sure all requirements above have been installed.
- follow the [SpotifyPlus Card Installation Instructions](https://github.com/thlucas1/spotifyplus_card/wiki/Installation-Instructions) to install the card.
- refer to the [Card Picker](https://github.com/thlucas1/spotifyplus_card/wiki/Configuration-Options#card-picker) topic for how to add the card to a dashboard; that document also covers the various card configuration options.

## More Information
Check out the following links for more information:

__Note that documentation is still a work in progress, and may be incomplete or even not applicable to the plugin.  I am using an existing plugin as a template for most of the docs, and have not gotten to some files yet to adjust the content.  The installation instructions and card configuration pages are done, but all others are work in progress.__

- [Card Wiki Home](https://github.com/thlucas1/spotifyplus_card/wiki)
- [Card Features](https://github.com/thlucas1/spotifyplus_card/wiki/Card-Features)
- [Card Configuration](https://github.com/thlucas1/spotifyplus_card/wiki/Configuration-Options)
- [Card UI Examples](https://github.com/thlucas1/spotifyplus_card/wiki/UI-Dashboards)
- [SpotifyPlus Integration](https://github.com/thlucas1/homeassistantcomponent_spotifyplus)

## Reporting a Problem
Submit a [Bug Report](https://github.com/thlucas1/spotifyplus_card/issues/new?assignees=&labels=bug&projects=&template=bug.yml) to bring the issue to my attention. I receive a notification when a new issue is opened, and will do my best to address it in a prompt and professional manner.

## Request a New Feature
Do you have an idea for a new feature that could be added to the dashboard?  Submit a [Feature Request](https://github.com/thlucas1/spotifyplus_card/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.yml) to get your idea into the queue. I receive a notification when a new request is opened, and will do my best to turn your idea into the latest and greatest feature.

**Comment by andrew-kennedy** [comment]
I would love to try it!

**Comment by ojbeltran** [comment]
Im ready....

**Reply by thlucas1** [reply]
I added a "Getting Started" section to the above text to better explain how to install the card.

**Comment by awsrossw** [comment]
I have the Card installed ! it is very nice work. I have played around with a few different configurations to display on my FireHD8 with Fully Kisosk browser.
I was looking around for a few things - since this is a touch interface, can we make the icons in the footer bigger? and the player controls.

Note: none of the sub sections work on the Fully Kiosk Browser - example the devices screen doesnt show anything but does report 7 items.

This is fantastic work!

**Comment by thlucas1** [comment]
@awsrossw 
Hi Ross - thanks for the kind words.

You mentioned "_Note: none of the sub sections work on the Fully Kiosk Browser - example the devices screen doesn't show anything but does report 7 items._" ... can you reply with a screen capture of what that looks like?  Just wondering why it doesn't show anything, but it has 7 items.  

Also, are you setting the card [`height`](https://github.com/thlucas1/spotifyplus_card/wiki/Configuration-Options#height) and [`width`](https://github.com/thlucas1/spotifyplus_card/wiki/Configuration-Options#width) configuration options?  If fully kiosk, you might want to use the following in your card configuration:
```
width: fill
height: fill
```

Thanks - Todd

**Comment by thlucas1** [comment]
@awsrossw 
Hi Ross - the footer icon size and player deck controls will be adjustable in v1.0.10 of the card (not released yet).

You mentioned "Note: none of the sub sections work on the Fully Kiosk Browser - example the devices screen doesn't show anything but does report 7 items." ... can you reply with a screen capture of what that looks like? Just wondering why it doesn't show anything, but it has 7 items.

Thanks - Todd

**Comment by thlucas1** [comment]
FYI - just released a new version of the [SpotifyPlus Card UI](https://github.com/thlucas1/spotifyplus_card/wiki).

[ 1.0.10 ] - 2024/11/03

  * This release requires the SpotifyPlus v1.0.64 release; please make sure you update the SpotifyPlus integration prior to updating this SpotifyPlus Card release.
  * Added `footerIconSize` general config option to change the size of the footer area icons.
  * Added `playerControlsIconSize` player controls config option to change the size of the player control area icons, volume mute icon, and power on/off icons.
  * Added actions dropdown menu to all section favorites browser details; most of these are the ability to search for related details.  More actions to come in future releases.
  * Added actions dropdown menu to all player information details; most of these are the ability to search for related details.  More actions to come in future releases.
  * Added ability to copy device details to the clipboard; for example, click on the value next to the `Device ID` title to copy the device id to the clipboard.
  * Added (all browsers) action: copy Spotify URI to clipboard.
  * Added playlist action: recover playlists via Spotify web ui.
  * Added playlist action: delete (unfollow) a playlist.
  * Updated playlist item action: track will now be added to the play queue instead of being played.  This will avoid wiping out the play queue.
  * Updated album item action: track will now be added to the play queue instead of being played.  This will avoid wiping out the play queue.
  * Updated podcast show item action: episode will now be added to the play queue instead of being played.  This will avoid wiping out the play queue.
  * Updated audiobook item action: chapter will now be added to the play queue instead of being played.  This will avoid wiping out the play queue.

**Comment by awsrossw** [comment]
Thank you - downloaded - will check it out.

**Comment by awsrossw** [comment]
Changing the icon sizes !  Very nice !!! more touch screen friendly.  Thankyou.
Sorry for the phone pics from the tablet
Nice big icons 5rem - on FireHD8
![IMG_8731](https://github.com/user-attachments/assets/39799e2c-8655-4054-86be-0440f74c750b)
Devices and playlists not working on FireHD8 - Fully Kiosk browser
![IMG_8732](https://github.com/user-attachments/assets/467850b9-827f-409f-9d68-a888848
![IMG_8733](https://github.com/user-attachments/assets/ee84036c-86f0-4f5a-a60e-dd4c552439f6)
5340e)

Verify on Desktop - 
![Screenshot 2024-11-04 at 8 36 18 AM](https://github.com/user-attachments/assets/8028a724-88a4-45cf-b090-e51d126cead1)
![Screenshot 2024-11-04 at 8 36 27 AM](https://github.com/user-attachments/assets/d2a6ea17-17ca-424d-aff2-3072858e058e)

Devices/playlists dont show up on HA App on Tablet either - Trying a different tablet.

**Reply by thlucas1** [reply]
@awsrossw 
Would you happen to know the screen size of the Fire HD tablet?  I will see if I can simulate your problem in Google Chrome developer tools, as I do not have a FireHD tablet.  It might also be related to a caching issue, though I doubt it as the bigger icons settings display properly.  To be safe, I would suggest clearing the browser cache on the FireHD tablet if you can.

Thanks for your feedback.

UPDATE - in researching this more, it appears the FireHD tablet supports different browsers.  Do you know which browser you are running for your kiosk, and what release version it is?  If available, can you try Google Chrome?  The display issue is probably related to a browser that does not support a certain level of CSS functions.  I can verify that though, if you can get me the browser type and it's version number.

**Comment by awsrossw** [comment]
FireHD8 - 7th Gen - 800 x 1280 pixels, 16:10 ratio
Using Fully Kiosk Browser - also tested Chrome - same results
Downloading Firefox - and will test. (update - didnt work either)

Powering up iPad also :) 

update: Im thinking it is the FireOS on the Kindle, im not able to root this version.

**Reply by thlucas1** [reply]
@awsrossw 
I was able to reproduce the behavior in Google Chrome developer tools.  
I also created issue [Fire HD8 Table not rendering media lists (awsrossw)](https://github.com/thlucas1/spotifyplus_card/issues/24) to track this, as I don't want to get the Beta discussion off topic.
Thanks.

**Comment by thlucas1** [comment]
The [SpotifyPlus Card Dashboard](https://community.home-assistant.io/t/spotifyplus-card-dashboard/795290) has been released, and the beta test program period is over.  

I appreciate everyone's help in the testing - it really helped, and I also gathered quite a few ideas that were (and will be) added.

Thanks again, and enjoy the card!

Todd

Change Log from final beta pre-release:

[ 1.0.12 ] - 2024/11/15

  * This release requires the SpotifyPlus v1.0.65 release; please make sure you update the SpotifyPlus integration prior to updating this SpotifyPlus Card release.
  * Added category browser: browse Spotify playlists by categories; existing card configurations have to enable the section in the general configuration settings.
  * Added dynamic track recommendation capability to user-defined presets. Simply put, you define a preset with the parameters of what you want to play and Spotify searches its media catalog for tracks that match. The matching tracks are then added to a play queue and played in random order. The matching tracks will change over time, as Spotify adds new content to its media catalog.
  * Added action for all playable media types: Copy Preset Info to Clipboard.  This will create a user-preset configuration entry for the selected media and copy it to the clipboard; the entry can then be pasted into the configuration editor under the `userPresets:` key, which will create a user preset for the media item.
  * Updated artist details to show more information about the artist.  Note that actions menu can be used to display more artist-related details (albums, top tracks, etc).
  * Added artist action: show artist albums; lists only the artist albums (no compilations, no appears on, no singles, etc).
  * Added artist action: show artist album compilations; lists only the artist compilation albums (no appears on, no singles, etc).
  * Added artist action: show artist albums appears on (aka collaborations); lists only the artist appears on albums (no compilations, no singles, etc).
  * Added artist action: show artist album singles; lists only the artist single release albums (no compilations, no appears on, etc).
  * Added artist action: show artist related artists; lists artists that are similar to the selected artist.
  * Added show action: search show episodes; lists show episodes with cover art for the selected show.
  * Updated show details form to only display the first 20 episodes of the show after the show description.  This will make the UI much more responsive, as most shows have 200+ episodes.  More shows can be listed by using the actions menu drop down.
